pkg/status/status.go

```diff
@@ -0,0 +1,183 @@
+package status
+
+import (
+	"context"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	cohelpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	operatorhelpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+	"github.com/operator-framework/operator-marketplace/version"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const clusterOperatorName = "marketplace-operator"
+
+type status struct {
+	configClient    *configclient.ConfigV1Client
+	coAPINotPresent bool
+	namespace       string
+	clusterOperator *configv1.ClusterOperator
+}
+
+// Status is the interface to report the health of the operator to CVO
+type Status interface {
+	SetFailing(message string)
+	SetAvailable(message string)
+}
+
+// New returns an initialized Status
+func New(cfg *rest.Config, mgr manager.Manager, namespace string) Status {
+	// The default client serves read requests from the cache which contains
+	// objects only from the namespace the operator is watching. Given we need
+	// to query CRDs which are cluster wide, we add the CRDs to the manager's
+	// scheme and create our own client.
+	if err := v1beta1.AddToScheme(mgr.GetScheme()); err != nil {
+		log.Fatal(err)
+	}
+
+	controllerClient, err := client.New(cfg, client.Options{Scheme: mgr.GetScheme()})
```

Comment:
If you use an openshift client `configclient.NewForConfig(cfg)` you don't see the expected `IsNoMatchError`?